### DISCUSSION TITLE
Test Suite Update

### DISCUSSION THREAD

**Discussion by MiguelMarcelino** [py2many/py2many#522] [Ideas]
I believe the test suite should get an update. It would be a good idea to integrate the [tests from py2cr](https://github.com/nanobowers/py2cr/tree/main/tests), as they are more isolated than the ones currently found in Py2Many.
I believe it would also be great to include a subset of Python's (more specifically, CPython's) test suite, to have a more official and robust test set.
Would like to hear your opinions on this.

**Comment by div72** [comment]
Seems like a good idea but wouldn't having that many tests make CI too slow?

**Reply by MiguelMarcelino** [reply]
That could be a problem. Although I believe we require smaller, more isolated tests. Maybe we could update the current regression tests instead.

**Comment by div72** [comment]
~~I've checked a few of CPython's tests and I think that porting them would require more effort than simply writing tests for the whole stdlib.~~ It might be feasible for some tests.

BTW py2cr also seem to have our tests: https://github.com/nanobowers/py2cr/tree/main/tests/py2many which is funny considering the idea of integrating their tests in my opinion. =)

**Reply by MiguelMarcelino** [reply]
I know. I suggested py2cr after reading [this](https://github.com/py2many/py2many/issues/415) discussion. I assume the author also did the same thing with our tests :).
The idea to integrate CPython's tests was to create a Rewriter that could translate them to more generic tests, although this ended up not being as straightforward.
